packages/cli-platform-android/src/commands/runAndroid/checkGradleWrapper.ts

```diff
@@ -0,0 +1,101 @@
+import path from 'path';
+import fs from 'fs';
+import {logger, prompt, version} from '@react-native-community/cli-tools';
+import chalk from 'chalk';
+import execa from 'execa';
+
+// TODO: Fetch it from somewhere?
+const gradleWrapperVersionsMap: Record<number, string> = {
```

Comment:
Don't really like this hardcoded array. Maybe we could fetch data about gradle versions from core somewhere? 
(But that would be time consuming and then we definitely should be running `checkGradleWrapper` on each run of command... ) 🤔

Reply:
In ideal world flow should look like this IMO:
1. We cache `package.json`,
2. While running `run/build-android` we check is it changes, if yes we do a request to GitHub and look for branch related to `react-native` version that is presented in `package.json`, then we do the logic and we run `./gradlew wrapper ...`

The drawback here is that we need to be sure the source of informations, if something change in future in response we potentially will need to release X patch releases.

Reply:
Can we just get it from the `react-native` package directly?

```
% cat node_modules/react-native/template/android/gradle/wrapper/gradle-wrapper.properties
distributionBase=GRADLE_USER_HOME
distributionPath=wrapper/dists
distributionUrl=https\://services.gradle.org/distributions/gradle-8.0.1-all.zip
networkTimeout=10000
zipStoreBase=GRADLE_USER_HOME
zipStorePath=wrapper/dists
```

Reply:
Also, I'm not sure this if this isn't overly aggressive. For instance, Gradle 7.5 works fine with 0.72, but with this change, you're forcing people to upgrade even though they might not be ready for that yet. We have a similar check in RNTA, but using hard-coded values because the template isn't always right: https://github.com/microsoft/react-native-test-app/blob/1289c283c034472be5adbdc33a2238fb08b27875/android/test-app-util.gradle#L9-L45

Reply:
> Also, I'm not sure this if this isn't overly aggressive. [...]

Maybe we can extend prompt question with "No, don't ask again" option and store that value in cache so we won't be asking repeatedly?